ethereum/statetransition/src/test/java/tech/pegasys/teku/statetransition/forkchoice/ForkChoiceBlobSidecarsAvailabilityCheckerTest.java

```diff
@@ -50,103 +68,284 @@ public class ForkChoiceBlobSidecarsAvailabilityCheckerTest {
+    final Optional<Throwable> cause =
+        failByException ? Optional.of(new RuntimeException("oops")) : Optional.empty();
 
-    when(miscHelpers.isDataAvailable(
-            eq(block.getSlot()),
-            eq(block.getRoot()),
-            argThat(new KzgCommitmentsArgumentMatcher(block)),
-            eq(blobSidecars)))
-        .thenThrow(cause);
+    final SafeFuture<BlobSidecarsAndValidationResult> availabilityCheckResult =
+        blobSidecarsAvailabilityChecker.getAvailabilityCheckResult();
+
+    SafeFutureAssert.assertThatSafeFuture(availabilityCheckResult).isNotCompleted();
 
+    // initiate availability check
     assertThat(blobSidecarsAvailabilityChecker.initiateDataAvailabilityCheck()).isTrue();
-    assertInvalid(blobSidecarsAvailabilityChecker.getAvailabilityCheckResult(), Optional.of(cause));
-    assertInvalid(blobSidecarsAvailabilityChecker.validate(blobSidecars), Optional.of(cause));
+
+    // mock kzg availability check failure for the initial set
+    if (failByException) {
+      whenDataAvailability(blobSidecarsInitial, kzgCommitmentsInitial).thenThrow(cause.get());
+    } else {
+      whenDataAvailability(blobSidecarsInitial, kzgCommitmentsInitial).thenReturn(false);
+    }
+
+    asyncRunner.executeDueActions();
+
+    assertInvalid(availabilityCheckResult, blobSidecarsInitial, cause);
+  }
+
+  @ParameterizedTest
+  @ValueSource(booleans = {true, false})
+  void shouldReturnNotAvailableIfSecondBatchFails(final boolean failByException) {
+    prepareInitialAvailability(Availability.PARTIAL);
+
+    final Optional<Throwable> cause =
+        failByException ? Optional.of(new RuntimeException("oops")) : Optional.empty();
+
+    final SafeFuture<BlobSidecarsAndValidationResult> availabilityCheckResult =
+        blobSidecarsAvailabilityChecker.getAvailabilityCheckResult();
+
+    SafeFutureAssert.assertThatSafeFuture(availabilityCheckResult).isNotCompleted();
+
+    // initiate availability check
+    assertThat(blobSidecarsAvailabilityChecker.initiateDataAvailabilityCheck()).isTrue();
+
+    // mock kzg availability check to be OK for the initial set
+    whenDataAvailability(blobSidecarsInitial, kzgCommitmentsInitial).thenReturn(true);
+
+    // let availability check to be performed.
+    asyncRunner.executeDueActions();
+
+    // mock kzg availability check failure for the initial set
+    if (failByException) {
+      whenDataAvailability(blobSidecarsAdditional, kzgCommitmentsAdditional).thenThrow(cause.get());
+    } else {
+      whenDataAvailability(blobSidecarsAdditional, kzgCommitmentsAdditional).thenReturn(false);
+    }
+
+    // let the tracker complete with all blobSidecars
+    completeTrackerWith(blobSidecarsComplete);
+
+    assertInvalid(availabilityCheckResult, blobSidecarsAdditional, cause);
   }
 
   @Test
-  void shouldReturnValid() {
-    prepareBlockAndBlobSidecarsInAvailabilityWindow(true);
+  void shouldReturnNotAvailableIfTrackerLiesWithCompletionButItIsNot() {
```

Comment:
I'd call it something like `shouldThrowWhen...` especially as you are not checking it with assertNotAvailable, but checking exception instead

Reply:
ah yes, it was the previous behaviour